Add BC, UO and misc doco to THM (#25116)
idaholab/moose: modules/thermal_hydraulics/doc/content/source/base/Simulation.md

# Simulation

> Simulation is one of the two base classes of a `THMProblem`. It implements a number of the specificities

Author: ```suggestion
Simulation is one of the two base classes of a [THMProblem.md]. It implements a number of the specificities
```

> of a thermal hydraulics solve, notably the reliance on `Components` to form the problem.

Author: ```suggestion
of a thermal hydraulics solve, notably the reliance on [Components](syntax/Components/index.md) to form the problem.
```

It is in charge of :

> - building the mesh, created by each [Component](syntax/Components/index.md)

Author: ```suggestion
- building the mesh, each part being created by each [Component](syntax/Components/index.md)
```

> - adding the variables

Author: ```suggestion
- creating the variables
```

> - setting initial conditions for the variables, whether from file or from user input

Author: ```suggestion
- setting up initial conditions for the variables, whether from file or from user input
```

- setting up the quadrature

It also has APIs, that are called by various actions when parsing the input file, to:

- add components
- initialize components
- add MOOSE objects created by components
- identify component loops
- set up the coordinate system

It also performs integrity checks for:

> - the coupling matrix, making sure it's sufficiently capture inter-variable dependencies

Author: ```suggestion
- the coupling matrix, making sure it sufficiently captures inter-variable dependencies
```
